projects/storefrontlib/src/cms-components/checkout/guards/checkout-details-loaded.guard.ts

```diff
@@ -0,0 +1,16 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, UrlTree } from '@angular/router';
+import { Observable } from 'rxjs';
+
+import { CheckoutDetailsService } from '../services/checkout-details.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CheckoutDetailsLoadedGuard implements CanActivate {
+  constructor(private checkoutDetailsService: CheckoutDetailsService) {}
+
+  canActivate(): Observable<boolean | UrlTree> {
```

Comment:
I am not quite sure I understand why this interceptor is necessary.

Reply:
Also the return type should just be `Observable<boolean>` as it will never return a UrlTree unless you want to add a redirect.

Reply:
for other steps, we have "xxx-set.guard", e.g. `shipping-address-set.guard.ts`. In these guards, the checkout details data is loaded. But, for the first step (shipping address step), we don't have such a guard for it. So, the cart's shipping address is not loaded. If you select an address, and reload the page, you will find that your selection is gone.

Reply:
Ah I see.